vite.config.ts

```diff
@@ -0,0 +1,37 @@
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+import visualizer from 'rollup-plugin-visualizer';
+import pkg from './package.json';
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [
+    react({
+        jsxImportSource: "@emotion/react",
+    }),
+  ],
+  build: {
+    outDir: "build",
+    lib: {
+      entry: "./src/App.tsx",
+      formats: ["cjs","es"],
+      fileName: (format) => {
+        const path = format === "es" ? pkg.module : pkg.main
```

Comment:
This ensures that the file name is coming from the `package.json`, rather than having a loose contract between it and the build config.

Added a comment in 9679bb7f608e993d1136504c96d685ce6531fb57.